server/models/PlatformSubscription.ts

```diff
@@ -94,6 +119,30 @@ class PlatformSubscription extends Model<
     return now.isSameOrBefore(this.endDate) && now.isSameOrAfter(this.startDate);
   }
 
+  prorateBasePrice(billingPeriod: BillingPeriod): number {
+    const billingStart = PlatformSubscription.periodStartDate(
+      PlatformSubscription.getBillingPeriodRange(billingPeriod),
+    );
+    const billingEnd = PlatformSubscription.periodEndDate(PlatformSubscription.getBillingPeriodRange(billingPeriod));
+
+    const billingTime = moment.utc(billingEnd).diff(billingStart, 'seconds');
```

Comment:
I think we should calculate base in a difference in days, but this is just my opinion, and we should check with product.